crates/storage/provider/src/post_state/mod.rs

```diff
@@ -308,7 +315,10 @@ impl PostState {
             our_storage.storage.extend(their_storage.storage);
         }
 
-        self.receipts.extend(other.receipts);
+        self.receipts.extend(
```

Comment:
I think such behaviour is a bit unobvious, I wouldn't expect that exactly on `expand` receipts will be filtered according to pruning settings.

I wonder if we should rather modify only the db writing part in `write_to_db`, skipping the receipts until some block number.

Reply:
That's what I had at first, but then I wanted to avoid the allocation of receipts which are not necessary. But I guess it is quite unobvious, reverting it back

Reply:
> then I wanted to avoid the allocation of receipts which are not necessary

I think we're already doing that because of receipts verification we need to do, and I'm not sure if there are any allocations on `extend` because it consumes the `receipts` iterator.